### PR TITLE
chore: make unified enginev3 ready

### PR DESCRIPTION
**Pull request by jonathanyeung**
## chore: make unified enginev3 ready

This change makes unified engine v3 compliant by implementing the pattern for parsing described in [[Engine-Less Workflow|dendron://dendron.docs/pkg.dendron-markdown.arch#engine-less-workflow]].  Notably, engine has been removed from all variations of `ProcOpts`. 

One other noteworthy change: If a wikilink doesn't have an explicit alias (such as `[[explicit alias | foo]]`), then the wikilink parser will leave the alias as `undefined` in the AST Node during parsing time instead of extrapolating an alias.  Any necessary extrapolation is now done during the compilation phase. 

### Known Regressions / Things that need to be fixed
- ~~wildcard wikilink references~~ Fixed 
- observing ancestor colors for hash tags

### CODE REVIEW

File: dendron-main.code-workspace [hunk 1]
```diff
@@ -103,7 +103,27 @@
...
+      "statusBar.foreground": "#e7e7e7",
+      "statusBarItem.hoverBackground": "#fa1b49",
+      "statusBarItem.remoteBackground": "#dd0531",
+      "statusBarItem.remoteForeground": "#e7e7e7",
+      "titleBar.activeBackground": "#dd0531",
+      "titleBar.activeForeground": "#e7e7e7",
+      "titleBar.inactiveBackground": "#dd053199",
+      "titleBar.inactiveForeground": "#e7e7e799"
+    },
+    "peacock.color": "#dd0531"
```
Comment (kevinslin) [inline dendron-main.code-workspace:126]: watch out for changing workspace settings

File: packages/common-all/src/dnode.ts [hunk 2]
```diff
@@ -1170,18 +1170,15 @@ export class NoteUtils {
...
-  }): {
+  static color(opts: { fname: string; vault?: DVault }): {
     color: string;
     type: "configured" | "generated";
   } {
-    const ancestors = NoteUtils.ancestors({ ...opts, includeSelf: true });
-    for (const note of ancestors) {
-      if (note.color) return { color: note.color, type: "configured" };
-    }
+    // JYTODO: Re-enable the ancestor color logic later
```
Comment (kevinslin) [inline packages/common-all/src/dnode.ts:1177]: just over communicating/confirming, were you planning on getting this wrapped up for this pr or in a separate pr? the enginev3 unification should not remove any existing feature from the desktop plugin

**Review by kevinslin — COMMENTED** [review]
was it necessary for all of this to be done in one pr? since this is such a large surface area, we should limit the changes. otherwise, it'll be hard to reconcile this with current code changes

File: packages/common-all/src/dnode.ts (hunk 2 above)
Comment (jonathanyeung) [reply packages/common-all/src/dnode.ts:1177]: Hadn't decided yet.  How important is this feature though? It seems quite arbitrary, and its implementation is not good from a performance point of view.

**Comment by jonathanyeung** [comment]
> it'll be hard to reconcile this with current code changes

This is actually why I want to do it in one go. Otherwise, migration will be a continually moving target and we won't finish.

File: packages/common-all/src/dnode.ts (hunk 2 above)
Comment (jonathanyeung) [reply packages/common-all/src/dnode.ts]: Did some more analysis, and unfortunately it's going to be quite difficult to re-implement this ancestor coloring feature.  IMO we should just remove it; it's not worth complicating the parsing logic.  I think it's debatable whether this behavior is desired anyway, and it probably won't affect many users.

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap [hunk 3]
```diff
@@ -35,12 +37,9 @@ VFile {
 exports[`GIVEN dendronPub (old tests - need to be migrated) enablePrettyRefs config.enablePrettyRef: false 1`] = `
 VFile {
   "contents": "<h1 id=\\"foo\\">Foo</h1>
-<p></p><p></p><p>bar body</p><p></p><p></p>
-<hr>
-<strong>Children</strong>
-<ol>
-<li><a href=\\"foo.ch1\\">Ch1</a></li>
-</ol>",
+<p></p><blockquote>
```
Comment (kevinslin) [inline packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:40]: same as above

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap [hunk 4]
```diff
@@ -77,12 +71,9 @@ VFile {
...
 VFile {
   "contents": "<h1 id=\\"foo\\"><a aria-hidden=\\"true\\" class=\\"anchor-heading icon-link\\" href=\\"#foo\\"></a>Foo</h1>
-<p></p><p></p><p>bar body</p><p></p><p></p>
-<hr>
-<strong>Children</strong>
-<ol>
-<li><a href=\\"/notes/foo.ch1\\">Ch1</a></li>
-</ol>",
+<p></p><blockquote>
+No note with name bar found in cache during parsing.
```
Comment (kevinslin) [inline packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:75]: missing bar note?

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap [hunk 5]
```diff
@@ -61,12 +60,7 @@ VFile {
 <div id=\\"portal-parent-anchor\\" class=\\"portal-parent\\" markdown=\\"1\\">
 <div class=\\"portal-parent-fader-top\\"></div>
 <div class=\\"portal-parent-fader-bottom\\"></div><p>bar body</p>
-</div></div><p></p><p></p>
-<hr>
-<strong>Children</strong>
-<ol>
-<li><a href=\\"foo.ch1\\">Ch1</a></li>
-</ol>",
+</div></div><p></p><p></p>",
```
Comment (kevinslin) [inline packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:63]: why did the children links dissapear?

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap (hunk 4 above)
Comment (kevinslin) [reply packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:75]: also no children links

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap [hunk 6]
```diff
@@ -103,12 +94,7 @@ VFile {
 <div id=\\"portal-parent-anchor\\" class=\\"portal-parent\\" markdown=\\"1\\">
 <div class=\\"portal-parent-fader-top\\"></div>
 <div class=\\"portal-parent-fader-bottom\\"></div><p>bar body</p>
-</div></div><p></p><p></p>
-<hr>
-<strong>Children</strong>
-<ol>
-<li><a href=\\"/notes/foo.ch1\\">Ch1</a></li>
-</ol>",
+</div></div><p></p><p></p>",
```
Comment (kevinslin) [inline packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:97]: apply above comments to other test cases

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/extendedImage.spec.ts.snap [hunk 7]
```diff
@@ -3,13 +3,7 @@
...
 VFile {
   "contents": "<h1 id=\\"root\\">Root</h1>
-<p><img src=\\"/assets/image.png\\" alt=\\"alt text\\" style=\\"width:50%;max-height:400px;opacity:0.8;\\"></p>
-<hr>
-<strong>Children</strong>
-<ol>
-<li><a href=\\"bar\\">Bar</a></li>
-<li><a href=\\"foo\\">Foo</a></li>
-</ol>",
+<p><img src=\\"/assets/image.png\\" alt=\\"alt text\\" style=\\"width:50%;max-height:400px;opacity:0.8;\\"></p>",
```
Comment (kevinslin) [inline packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/extendedImage.spec.ts.snap:6]: missing children links

Comment (kevinslin) [reply packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/extendedImage.spec.ts.snap:6]: apply to other test cases here as well

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/noteRefv2.spec.ts.snap [hunk 8]
```diff
@@ -49,7 +49,7 @@ exports[`noteRefV2 WHEN parse begin block AND note starts with a header "MD_REGU
 VFile {
   "contents": "# Foo
 
-
+> No note with name foo found in cache during parsing.
```
Comment (kevinslin) [inline packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/noteRefv2.spec.ts.snap:53]: ?

**Review by kevinslin — COMMENTED** [review]
some comments on the snapshot tests

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap [hunk 9]
```diff
@@ -24,7 +24,9 @@ VFile {
 exports[`GIVEN dendronPub (old tests - need to be migrated) enablePrettyRefs WHEN config sets enablePrettyRefs true AND note overrides to false THEN renders without pretty refs 1`] = `
 VFile {
   "contents": "<h1 id=\\"with-override\\"><a aria-hidden=\\"true\\" class=\\"anchor-heading icon-link\\" href=\\"#with-override\\"></a>With Override</h1>
-<p></p><p></p><p>bar body</p><p></p><p></p>",
+<p></p><blockquote>
```
[the remark this answers is not in the document]
Comment (jonathanyeung) [reply packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap]: Fixed + changed the validation to actually check for bar body.

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap [hunk 10]
```diff
@@ -536,7 +536,7 @@ exports[`GIVEN dendronPub (old tests - need to be migrated) note reference fail:
 VFile {
   "contents": "<h1 id=\\"ref\\">Ref</h1>
 <p></p><blockquote>
-No note with name bar found in vault vault2
+No note with name bar found in cache during parsing.
```
Comment (kevinslin) [inline packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/dendronPub.spec.ts.snap:539]: for a separate pr - can we have a more detailed error message here? (include vault name)

File: packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/extendedImage.spec.ts.snap (hunk 7 above)
Comment (kevinslin) [reply packages/engine-test-utils/src/__tests__/engine-server/markdown/__snapshots__/extendedImage.spec.ts.snap:6]: this still seems to be the case?

**Pull request merged**